src/batSplay.ml

```diff
@@ -769,6 +794,37 @@ struct
     | Empty -> raise Not_found
     | Node (l, kv, r) -> kv, sref (bst_append l r)
 
+
+  let add_seq s m =
+    BatSeq.fold_left
+      (fun m (k, v) -> add k v m)
+      m
+      s
+    
+  let of_seq s =
+    add_seq s empty
+
+  let rec to_seq m =
+    fun () ->
+    (* since the tree can change shape arbitrarily, fetch all bindings and turn them into a seq. *)
+    BatSeq.of_list (bindings m) ()
+    
+  let rec to_seq_from k m =
+    fun () ->
+    BatSeq.filter (fun (k2, _) -> Ord.compare k k2 <= 0) (to_seq m) ()
```

Comment:
Again, this implementation is not satisfactory, it does not provide the expected complexity. Could you try implementing this by using a function that takes a bound `k` and a search tree `tr`, and returns an `'a enumeration` of the elements whose keys are greater or equal to `k`?

Reply:
Yes, i will re-use the implementation from BatMap.

Reply:
... is now implemented in terms of 'a enumeration